### ISSUE TITLE
Unicode Issue on Path Completion

### ISSUE THREAD

**Issue by mitnk** [murarth/linefeed#54]
I encountered that linefeed panic when do path completion. I checked and found `longest_common_prefix()` does not consider unicode chars. `cargo test` would fail with following cases:

```
        # case 1 - fail
        assert_eq!(longest_common_prefix(["hi我的123", "hi我的134"].iter()),
            Some("hi我的1"));
        # case 2 - panic
        assert_eq!(longest_common_prefix(["hi我的世界123", "hi我的世界134"].iter()),
            Some("hi我的世界1"));
```

**Comment by murarth** [comment]
Just pushed a commit to fix this. Thanks for the issue.

**Comment by mitnk** [comment]
Thanks a lot

**Comment by mitnk** [comment]
Hi @murarth, would you mind to release a new version about this when you got time? :)

**Comment by murarth** [comment]
@mitnk: Just published 0.5.4
